src/pages/home/library/Discography.tsx

```diff
@@ -32,6 +35,23 @@ const Discography: FunctionComponent = () => {
       );
     }
   };
+
+  // Keep song in a playing state till the song has been filtered out
+  useEffect(() => {
+    let isSongFound = false;
+    if (currentPlayingSong !== null && filteredData !== undefined)
+      filteredData.forEach((filteredSong) => {
+        if (isSongFound !== true && filteredSong.id === currentPlayingSong.id) {
+          isSongFound = true;
+        }
+      });
+    else {
+      isSongFound = true;
+    }
+
+    isSongFound ? null : setCurrentPlayingSong(null);
```

Comment:
Maybe change this to something like
```
if (!isSongFound) {
  setCurrentPlayingSong(null);
}
```
I feel it's a bit strange to use the ternary operator in this way.

Reply:
My bad on the ternary operator, definitely something I need to unlearn. I believe the biggest issue while working through this, and why my logic was a little convoluted, was making the flow work with initial song querying and filtering.

I added the following to update the `filteredData` once the song query, `useGetSongsQuery()`, completes so that when the songs appear in the UI `isSongFound` works properly, before `filteredData` would just start with an empty array and cause havoc on the play pause functionality. If it doesn't make sense to update `filteredData` at the start of the page load like this please let me know.
```
useEffect(() => {
    setFilteredData(songData);
  }, [songData]);
```